pkg/networkservice/common/retry/client.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2021 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package retry provides a networkservice.NetworksrviceClient wrapper that allows to retries requests and closes.
+package retry
+
+import (
+	"context"
+	"time"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/clock"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
+// Settings represents retry policy settings.
+type Settings struct {
+	Timeout    time.Duration
+	Interval   time.Duration
+	TryTimeout time.Duration
+}
+
+type retryClient struct {
+	Settings
+	client networkservice.NetworkServiceClient
+}
+
+// Option configuress retry.Client instance.
+type Option func(*retryClient)
+
+// WithSettings sets retry policy settings for the retry.Client instance.
+func WithSettings(s Settings) Option {
```

Comment:
Why not do this as standard options like 

retry.WithInterval
retry.WithTryTimeout

instead of 'Settings'

Reply:
Thanks, fixed.